.travis.yml

```diff
@@ -1,12 +1,31 @@
-script:
-    - docker build --target test --tag my-test-image .
-    - docker run --env-file .env.test my-test-image tests
-    - docker run --env SECRET_KEY --env TRELLO_KEY --env TRELLO_TOKEN my-test-image tests_e2e
+jobs:
+  include:
+    - stage: build docker test image and run tests
+      script:
+        - docker build --target test --tag my-test-image .
+        - docker run --env-file .env.test my-test-image tests
+        - docker run --env SECRET_KEY --env TRELLO_KEY --env TRELLO_TOKEN my-test-image tests_e2e
+    - stage: build docker production image and push to docker hub
+      script:
+        - echo "$DOCKER_HUB_PASSWORD" | docker login -u "$DOCKER_HUB_USERNAME" --password-stdin
+        - docker build --target production --tag $DOCKER_HUB_USERNAME/todo:$TRAVIS_COMMIT .
+        - docker push $DOCKER_HUB_USERNAME/todo:$TRAVIS_COMMIT
+    - stage: deploy production image to heroku
```

Comment:
The previous stage running on every push/PR is reasonable since you're only pushing a specific commit image, but we'd definitely want to restrict actual deployments to branches that have been reviewed & merged, whereas this is deploying immediately (good for testing but not longer term!)
We can set specific conditions on this script, but probably nicer is to take advantage of the default `before_deploy` and `deploy` lifecycle stages in Travis. Sadly, this is all academic as I can see you've been hit by the free trial expiry date and we'll need to switch over to Github Actions. I'll add an example workflow to get you started as you've already done the hard work here!